mpi-proxy-split/test/run_mana_mpi_regression_test.py

```diff
@@ -0,0 +1,206 @@
+    run_child = subprocess.Popen([cmd], shell=True)
+
+    # Wait for ranks to start
+    while get_running_ranks(mana_dir, "WorkerState::RUNNING") != int(ranks):
+        sleep(0.5)
+
+    print("Running")
+
+    # Test executable starting, checkpoint 3 times
+    eprint("Test started...")
+    for i in range(3):
+        time.sleep(3)
+        if run_child.poll() is None:
+            # Send checkpoint command and wait on it to succeed
+            eprint("Sending checkpoint command...")
+            ckpt_child = subprocess.run([f'{mana_dir}/bin/mana_status',
+                                         '--checkpoint'])
+            if ckpt_child.check_returncode() is not None:
+                eprint("Test case failed: checkpointing failed")
+                return 1
+            eprint("Checkpoint command successful")
+
+            # Wait for checkpoint to complete, make sure no ranks die
+            while get_running_ranks(mana_dir, "WorkerState::RUNNING") \
+                    != int(ranks):
+                if get_running_ranks(mana_dir) != int(ranks):
+                    eprint("Test case failed: not enough \
+                        ranks running while checkpointing")
+                    return 6
+
+            # Kill DMTCP Coord
+            kill_child = subprocess.Popen(['pkill', '-9', 'dmtcp_coord'])
+            kill_child.wait()
+
+            # Wait for DMTCP Coord to die, then restart it
+            status = subprocess.run([f'{mana_dir}/bin/mana_status'],
+                                     stdout=PIPE, stderr=subprocess.DEVNULL)
+            while "Checking for coordinator" \
+                    not in status.stdout.decode('utf-8'):
+                status = \
+                    subprocess.run([f'{mana_dir}/bin/mana_status'],
+                                    stdout=PIPE, stderr=PIPE)
+            eprint("Checkpointed, restarting...")
+            subprocess.run([f"{mana_dir}/bin/mana_coordinator"],
+                           stderr=subprocess.DEVNULL, stdout=subprocess.DEVNULL)
+
+            # Wait for coord to restart, then restart ranks
+            time.sleep(1)
+            run_child = subprocess.Popen([f'{restart}', '-n', f'{ranks}',
+                                         f'{mana_dir}/bin/mana_restart'],
+                                         stdout=subprocess.DEVNULL,
+                                         stderr=subprocess.DEVNULL)
+
+            # Wait for ranks to restart without dying
+            while get_running_ranks(mana_dir, "WorkerState::RUNNING") \
+                    != int(ranks):
+                if get_running_ranks(mana_dir) \
+                        != int(ranks) and \
+                        get_running_ranks(mana_dir, "WorkerState::RESTARTING")\
+                        == 0:
+
+                    eprint("Test case failed: not enough \
+                        ranks running after restart")
+                    return 2
+
+            time.sleep(3)
+
+            clean()
```

Comment:
Let's add a comment stating that "Clean up checkpoint images and restart scripts"